lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -0,0 +1,244 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from collections import namedtuple
+from ryu.ofproto.ofproto_v1_4 import OFPP_LOCAL
+
+from .base import MagmaController
+from magma.pipelined.openflow import flows
+from magma.pipelined.openflow.magma_match import MagmaMatch
+from magma.pipelined.app.inout import INGRESS
+from ryu.lib.packet import ether_types
+from magma.pipelined.app.base import MagmaController, ControllerType
+from magma.pipelined.utils import Utils
+
+GTP_PORT_MAC = "02:00:00:00:00:01"
+
+class Classifier(MagmaController):
+    """
+    A controller that sets up an openflow pipeline for Magma.
+    This controller is used for install the default tunnel entry,
+    add tunnel flows for gtp_port/local_port/mtr_port and delete
+    tunnel flows for gtp_port/local_port/mtr_port into OVS table 0.
+    """
+    APP_NAME = "classifier"
+    APP_TYPE = ControllerType.SPECIAL
+    ClassifierConfig = namedtuple(
+            'ClassifierConfig',
+            ['gtp_port', 'uplink_port_name', 'mtr_ip', 'mtr_port', 'internal_sampling_port', 'internal_sampling_fwd_tbl'],
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(Classifier, self).__init__(*args, **kwargs)
+        self.config = self._get_config(kwargs['config'])
+        self.tbl_num = self._service_manager.get_table_num(self.APP_NAME)
+        self.next_table = self._service_manager.get_table_num(INGRESS)
+        self._uplink_port = OFPP_LOCAL
+        self._datapath = None
+
+    def _get_config(self, config_dict):
+        port_name = None
+        mtr_ip = None
+        mtr_port = None
+        
+        if 'ovs_uplink_port_name' in config_dict:
+            port_name = config_dict['ovs_uplink_port_name']
+
+        if 'mtr_ip' in config_dict:
+            self._mtr_service_enabled = True
+            mtr_ip = config_dict['mtr_ip']
+            mtr_port = config_dict['ovs_mtr_port_number']
+
+        return self.ClassifierConfig(
+            gtp_port=config_dict['ovs_gtp_port_number'],
+            uplink_port_name=port_name,
+            mtr_ip=mtr_ip,
+            mtr_port=mtr_port,
+            internal_sampling_port=
+                               config_dict['ovs_internal_sampling_port_number'],
+            internal_sampling_fwd_tbl=
+                              config_dict['ovs_internal_sampling_fwd_tbl_number'],
+
+        )
+
+    def initialize_on_connect(self, datapath):
+        self._datapath = datapath
+        self._delete_all_flows()
+        self._install_default_tunnel_flows()
+        self._install_internal_pkt_fwd_flow()
+
+    def _delete_all_flows(self):
+        flows.delete_all_flows_from_table(self._datapath, self.tbl_num)
+
+    def cleanup_on_disconnect(self, datapath):
+        self._delete_all_flows()
+
+    def convert_precedence_to_priority(self, precedence:int):
```

Comment:
can you remove this function, since it is not used anywhere.

Reply:
Yes, will remove now. I forgot to remove this un-used function.